packages/phone-number-privacy/monitor/src/test.ts

```diff
@@ -86,38 +86,66 @@ export async function concurrentRPSLoadTest(
     | CombinerEndpointPNP.PNP_QUOTA
     | CombinerEndpointPNP.PNP_SIGN = CombinerEndpointPNP.PNP_SIGN,
   duration: number = 0,
-  bypassQuota?: boolean,
-  useDEK?: boolean
+  bypassQuota: boolean = false,
+  useDEK: boolean = false,
+  movingAverageRequests: number = 50
 ) {
-  const taskFn = async (i: number) => {
-    const start = performance.now()
+  const latencyQueue: number[] = []
+  let movingAvgLatencySum = 0
+  let latencySum = 0
+  let index = 1
+
+  function measureLatency(fn: () => Promise<void>): () => Promise<void> {
+    return async () => {
+      const start = performance.now()
+
+      await fn()
+
+      const reqLatency = performance.now() - start
+      latencySum += reqLatency
```

Comment:
i was originally imaginging that this logic lives in the requestEnder, so `taskFn` returns ` Promise<duration>` and the request ender is computing all of this in a single place. Which sounded better that every task updates the value.